### ISSUE TITLE
Make ion_fraction fields from existing ion_density fields if possible

### ISSUE THREAD

**Issue by chummels** [trident-project/trident#56]
Many codes already have certain ion density and ion number density fields created for H and He species because they're generated in the simulation.  When trident is asked to generate H and He fields for such a dataset, it (correctly) chooses to not overwrite these fields unless forced.  However, it adds the undefined `H_p0_ion_fraction` and `H_p1_ion_fraction` fields using its own calculation of those from the `ion_fraction` tables, instead of using the already generated `H_p0_density` and `H_p1_density` fields.  This leads to these the `ion_fraction` fields being inconsistent with the `density` and `number_density` fields.

**Comment by chummels** [comment]
```
import yt
import trident
ds = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
trident.add_ion_fields(ds, 'H')
ad = ds.all_data()

# True H I ion fraction from the already-defined number density fractions
print(ad['H_p0_number_density'] / ad['H_nuclei_density'])

# What Trident thinks is the H I ion fraction field
print(ad['H_p0_ion_fraction'])
```

Returns:

```
[  1.09266207e-06   1.09389294e-06   1.09390709e-06 ...,   5.38337805e-01
   6.33352805e-01   9.50010426e-01] dimensionless
[ 0.9999799   0.99988538  0.99991184 ...,  0.98312412  0.98809352
  0.99770006] dimensionless
```

When it *should* return the first array twice.

**Comment by chummels** [comment]
Addressed in PR #64
